components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -217,7 +225,24 @@ public static IdentityProvider buildFederatedIdentityProvider(HttpServletRequest
             }
 
             if (oldIdentityProvider != null && oldIdentityProvider.getCertificate() != null) {
-                paramMap.put("oldCertFile", oldIdentityProvider.getCertificate());
+                if (oldIdentityProvider.getCertificateInfoArray() != null && oldIdentityProvider.
+                        getCertificateInfoArray().length > 1) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("More than one certificate has been found as old certificate.");
+                    }
+
+                    StringBuilder multipleCertificate = new StringBuilder();
+                    for (CertificateInfo certificateInfo : oldIdentityProvider.getCertificateInfoArray()) {
+                        multipleCertificate.append(new String(Base64.decode(certificateInfo.getCertValue())));
+                    }
+                    paramMap.put(IdentityApplicationConstants.OLD_CERT_FILE, Base64.encode(multipleCertificate.toString().
+                            getBytes("UTF-8")));
```

Comment:
shall we use the getBytes method that accept Charset, instead of charset name.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/1704/commits/cdea1a45f0fde05acfbca6400cf8c41db1ee6493